tutorial/westeros/westeros.ipynb

```diff
@@ -0,0 +1,1268 @@
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'mode': 'standard',\n",
+    "    'time': 'year',\n",
+    "    'unit': 'USD/GWa',\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# in $ / MWh\n",
+    "costs = {\n",
+    "    'coal_ppl': 24.4,\n",
+    "    'grid': 47.8,\n",
+    "}\n",
+    "\n",
+    "for tec, val in costs.items():\n",
+    "    df = make_df(base_var_cost, technology=tec, value=val * 8760. * 1e3)\n",
+    "    scenario.add_par('var_cost', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "A full model will also have costs associated with\n",
+    "\n",
+    "- resource extraction: $\\sum_{c,g} \\ resource\\_cost_{n,c,g,y} \\cdot EXT_{n,c,g,y} $\n",
+    "- emissions (TODO recopy this, hats no good): $\\sum_{e,t, e \\in E(e)} emission\\_scaling_{e,e} \\cdot \\ emission\\_tax_{n,e,t,y} \\cdot EMISS_{n,e,t,y}$\n",
+    "- land use (emulator): $\\sum_{s} land\\_cost_{n,s,y} \\cdot LAND_{n,s,y}$\n",
+    "- exceedence on \"hard\" bounds of technology expansion and contraction: $\\sum_{m,h} \\ \\Big( abs\\_cost\\_activity\\_soft\\_lo_{n,t,y,m,h} + level\\_cost\\_activity\\_soft\\_lo_{n,t,y,m,h} \\cdot\\ levelized\\_cost_{n,t,y,m,h}\\Big) \\cdot ACT\\_LO_{n,t,y,h}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Additional Parameters\n",
+    "\n",
+    "In order for the mechanics of the model to work, there are some additional parameters that need to be defined. Specifically in this case, we need to provide `technical_lifetime`s for our various energy technologies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "base_technical_lifetime = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': horizon,\n",
+    "    'unit': 'y',\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "lifetimes = {\n",
+    "    'coal_ppl': 20,\n",
+    "    'wind_ppl': 10,\n",
+    "    'bulb': 1,\n",
+    "}\n",
+    "\n",
+    "for tec, val in lifetimes.items():\n",
+    "    df = make_df(base_technical_lifetime, technology=tec, value=val)\n",
+    "    scenario.add_par('technical_lifetime', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Commit the datastructure and solve the model"
```

Comment:
Legacy term 'datastructure', should be 'scenario'

Reply:
done!